src/Types/ScriptLookups.purs

```diff
@@ -987,68 +978,37 @@ processConstraint mpsMap osMap = do
       networkId <- getNetworkId
       let amount = fromPlutusValue plutusValue
       runExceptT do
-        -- If datum is presented, add it to 'datumWitnesses' and Array of datums.
-        -- Otherwise continue, hence `liftEither $ Right unit`.
-        maybe (liftEither $ Right unit) (ExceptT <<< addDatum) mDatum
-        let
-          liftDatumHash
-            :: forall (e :: Type) (dh :: Type)
-             . e
-            -> Maybe dh
-            -> Either e (Maybe dh)
-          liftDatumHash e Nothing = Left e
-          liftDatumHash _ (Just x) = Right (Just x)
-        -- [DataHash Note]
```

Comment:
I've dropped this note as it doesn't really apply anymore, due to us no longer using `Maybe Datum`, but a custom sum type.